src/main/java/org/subethamail/smtp/server/SourceTrackingSessionHandler.java

```diff
@@ -0,0 +1,96 @@
+package org.subethamail.smtp.server;
+
+import java.net.InetAddress;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * A {@link SessionHandler} to track and limit connection counts by remote addresses.
+ *
+ * @author Diego Salvi
+ */
+public class SourceTrackingSessionHandler implements SessionHandler {
```

Comment:
Make class final because in public API and we don't want to have to support inheritance. If people want to reuse then they can use composition.